modules/imgproc/include/opencv2/imgproc.hpp

```diff
@@ -4929,10 +4929,12 @@ Point LineIterator::pos() const
 
 //! @endcond
 
+CV_EXPORTS_W float local_cost(const Point& p, const Point& q, const Mat& gradient_magnitude, const Mat& Iy, const Mat& Ix, const Mat& zero_crossing);
+
+CV_EXPORTS_W void find_min_path(const Point& start, Mat img, Mat zero_crossing, Mat gradient_magnitude, Mat Ix, Mat Iy, Mat& hit_map_x, Mat& hit_map_y);
```

Comment:
This should be a class which computes all the intermediate buffers inside.

Reply:
Ok

Reply:
Agreed.
`imgproc` module location is fine.
But we need to define "Algorithm" class for this task (see `QRCodeDetector` as reference).
API should be clear, simple and easy to use (instead of local functions with many parameters which are hard to use).
Also add documentation for that.